Tests/Map.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if !defined(SKIP_INTERNAL_TESTS) || defined(RUN_COMMONTYPE_TESTS) || defined(RUN_COMMONTYPE_MAP_TESTS)
+
+	#include "CSP/Common/Map.h"
+
+	#include "CSP/Common/Optional.h"
+	#include "TestHelpers.h"
+
+	#include <gtest/gtest.h>
+
+using namespace csp::common;
+
+// Test fixture for csp::common::Map testing
+class MapTest
+{
+public:
+	MapTest()
+	{
+		MyMap[1] = "One";
+		MyMap[2] = "Two";
+		MyMap[3] = "Three";
+	};
+
+	Map<int, csp::common::String> MyMap;
+};
```

Comment:
We should try to adhere to the same testing style in all of our tests. Other csp::common tests create the test object locally. Relying on a class having the test object as a field changes the lifetime and behaviour of said test object.